src/client/automation/playback/drag/to-element.js

```diff
@@ -1,25 +1,28 @@
 import testCafeCore from '../../deps/testcafe-core';
 import DragAutomationBase from './base';
-import { getDefaultAutomationOffsets } from '../../utils/offsets';
+import { getOffsetOptions } from '../../utils/offsets';
 
 var positionUtils = testCafeCore.positionUtils;
 
 
 export default class DragToElementAutomation extends DragAutomationBase {
-    constructor (element, destinationElement, mouseOptions) {
-        super(element, mouseOptions);
+    constructor (element, destinationElement, dragToElementOptions) {
+        super(element, dragToElementOptions);
 
         this.destinationElement = destinationElement;
-    }
-
-    _getEndPoint () {
-        return positionUtils.findCenter(this.destinationElement);
+        this.destinationOffsetX = dragToElementOptions.destinationOffsetX;
+        this.destinationOffsetY = dragToElementOptions.destinationOffsetY;
     }
 
     _getDestination () {
-        var element = this.destinationElement;
-        var offsets = getDefaultAutomationOffsets(this.destinationElement);
+        var element     = this.destinationElement;
+        var elementRect = positionUtils.getElementRectangle(element);
+        var offsets     = getOffsetOptions(element, this.destinationOffsetX, this.destinationOffsetY);
+        var endPoint    = {
```

Comment:
AFAIR there was a style rule to separate multiline decalrations with a one blank line.

Reply:
yep